tests/runner/src/env.rs

```diff
@@ -52,8 +54,46 @@ impl EnvController for Env {
 
     async fn start(&self, mode: &str, _config: Option<&Path>) -> Self::DB {
         match mode {
-            "standalone" => self.start_standalone().await,
-            "distributed" => self.start_distributed().await,
+            "standalone" => {
+                if let Some(server_addr) = self.server_addr.clone() {
+                    let client = Client::with_urls(vec![server_addr.clone()]);
+                    let db = DB::new(DEFAULT_CATALOG_NAME, DEFAULT_SCHEMA_NAME, client);
+                    GreptimeDB {
+                        client: TokioMutex::new(db),
+                        server_processes: None,
+                        metasrv_process: None,
+                        frontend_process: None,
+                        ctx: GreptimeDBContext {
+                            time: 0,
+                            datanode_id: Default::default(),
+                        },
+                        is_standalone: false,
+                        env: self.clone(),
+                    }
```

Comment:
Can we refactor the function `start_standalone` and `start_distributed` to make them accept `&Option<String>` which represents the optional server address? Those logics look the same but with a few modifications.

Reply:
If so, we can just change this into:

```rust
self.start_standalone(&start_distributed).await;
```

Reply:
The `Env` struct already has the `server_addr` field, and I put the duplicated logic in a new function `connect_db()`.